docs/tutorials/using-backstage-proxy-within-plugin.md

```diff
@@ -0,0 +1,210 @@
+`X-Custom-Source`. You will need to add the following to `app-config.yaml`:
+
+```yaml
+proxy:
+  '/<your-proxy-uri>':
+    target: https://api.myawesomeservice.com/v1
+    changeOrigin: true # Use this to avoid cors related issues
+    headers:
+      X-Custom-Source: backstage
+```
+
+You can find more details about the proxy config options in the
+[proxying section](../plugins/proxying.md).
+
+# Calling an API using the backstage proxy
+
+If you followed the previous steps, you should now be able to access your API by
+calling `${backend-url}/api/proxy/<your-proxy-uri>`. The reason why
+`backend-url` is referenced is because the backstage backend creates and runs
+the proxy. Backstage is structured in such a way that you could run the
+backstage frontend independently of the backend. So when calling your API you
+need to prepend the backend url to your http call. The supported way of doing
+this, is to use [`createApiRef`](../reference/core-plugin-api.createapiref.md)
+to create a new [`ApiRef`](../reference/core-plugin-api.apiref.md) and register
+an [`ApiFactory`](../reference/core-plugin-api.apifactory.md), which will wrap
+your API implementation, with your plugin using
+[`createApiFactory`](../reference/core-plugin-api.createapifactory.md). Then,
+you can use your API in your components by calling
+[`useApi`](../reference/core-plugin-api.useapi.md)
+
+## Defining the API client interface
+
+Continuing from the previous example, let's assume that
+_https://api.myawesomeservice.com/v1_ has the following endpoints:
+
+| Method                   | Description             |
+| :----------------------- | :---------------------- |
+| `GET /users`             | Returns a list of users |
+| `GET /users/{userId}`    | Returns a single user   |
+| `DELETE /users/{userId}` | Deletes a user          |
+
+Here is an example definition for this API following backstage's `apiRef` style:
+
+```ts
+/* src/api.ts */
+import { createApiRef } from '@backstage/core-plugin-api';
+
+export interface User {
+  name: string;
+  email: string;
+}
+
+export interface MyAwesomeApi {
+  url: string;
+  listUsers: () => Promise<List<User>>;
+  getUser: (userId: string) => Promise<User>;
+  deleteUser: (userId: string) => Promise<boolean>;
+}
+
+export const myAwesomeApiRef = createApiRef<MyAwesomeApi>({
+  id: 'plugin.my-awesome-api.service',
+  description: 'Example API definition',
+});
+```
+
+## Creating the API client
+
+The `myAwesomeApiRef` is what you will use within backstage to reference the API
+client in your plugin. The API ref itself is a global singleton object that
+allows you to reference your instantiated API. The actual implementation would
+look something like this:
+
+```ts
+/* src/api.ts */
+
+/* ... */
+
+import { Config } from '@backstage/config';
+
+export class MyAwesomeApiClient implements MyAwesomeApi {
+  static fromConfig(config: Config) {
+    return new MyAwesomeApiClient(config.getString('backend.baseUrl'));
```

Comment:
Use the `DiscoveryApi` instead! See for example https://github.com/backstage/backstage/blob/2751a24b7d00716ac3c7172a59e55d3517e6ab34/plugins/fossa/src/api/FossaClient.ts#L59